nsv13/code/modules/mob/living/carbon/human/species_types/dwarves.dm

```diff
@@ -0,0 +1,223 @@
+/datum/species/dwarf/proc/handle_speech(datum/source, list/speech_args)
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = " [message]" //Credits to goonstation for the strings list.
+		var/list/dwarf_words = strings("dwarf_replacement.json", "dwarf") //thanks to regex too.
+
+		for(var/key in dwarf_words) //Theres like 1459 words or something man.
+			var/value = dwarf_words[key] //Thus they will always be in character.
+			if(islist(value)) //Whether they like it or not.
+				value = pick(value) //This could be drastically reduced if needed though.
+
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
+			message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
+
+		if(prob(3))
+			message += pick(" By Armok!")
+	speech_args[SPEECH_MESSAGE] = trim(message)
+
+//This mostly exists because my testdwarf's liver died while trying to also not die due to no alcohol.
+/obj/item/organ/liver/dwarf
+	name = "dwarf liver"
+	icon_state = "liver"
+	desc = "A dwarven liver, theres something magical about seeing one of these up close."
+	alcohol_tolerance = 0 //dwarves really shouldn't be dying to alcohol.
+	toxTolerance = 5 //Shrugs off 5 units of toxins damage.
+	maxHealth = 150 //More health than the average liver, as you aren't going to be replacing this.
+	//If it does need replaced with a standard human liver, prepare for hell.
+
+/obj/item/organ/dwarfgland //alcohol gland
+	name = "dwarf alcohol gland"
+	icon_state = "plasma" //Yes this is a actual icon in icons/obj/surgery.dmi
+	desc = "A genetically engineered gland which is hopefully a step forward for humanity."
+	w_class = WEIGHT_CLASS_NORMAL
+	var/stored_alcohol = 250 //They start with 250 units, that ticks down and eventaully bad effects occur
+	var/max_alcohol = 500 //Max they can attain, easier than you think to OD on alcohol.
+	var/heal_rate = 0.5 //The rate they heal damages over 400 alcohol stored
+	var/alcohol_rate = 10 //Its times 0.025 making it tick down by .25 per loop.
+
+/obj/item/organ/dwarfgland/prepare_eat()
+	var/obj/S = ..()
+	S.reagents.add_reagent("ethanol", stored_alcohol/10)
+	return S
+
+/obj/item/organ/dwarfgland/on_life()
+	//Filth Reactions - Since miasma now exists
+	var/filth_counter = 0
+	for(var/fuck in view(owner,7)) //hello byond for view loop, luckily its a custom organ.
+		if(istype(fuck, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = fuck
+			if(H.stat == DEAD)
+				filth_counter += 10
+		if(istype(fuck, /obj/effect/decal/cleanable/blood))
+			if(istype(fuck, /obj/effect/decal/cleanable/blood/gibs))
+				filth_counter += 1
+			else
+				filth_counter += 0.1
+		if(istype(fuck,/obj/effect/decal/cleanable/vomit)) //They are disgusted by their own vomit too.
+			filth_counter += 10 //Dwarves could technically chainstun each other in a vomit tantrum spiral.
+	switch(filth_counter)
```

Comment:
Is this not going to run per-tick, meaning that they get spammed with messages about untidiness?

Reply:
I'm going to redo a large chunk of this entire section later to try to optimize it.

Reply:
I just redid the on_life segments.